src/vmm/src/lib.rs

```diff
@@ -157,10 +168,25 @@ impl TryFrom<VMMConfig> for VMM {
             device_mgr: Arc::new(Mutex::new(IoManager::new())),
             event_mgr: EventManager::new().map_err(Error::EventManager)?,
             kernel_cfg: config.kernel_config.clone(),
+            block_devices: Vec::new(),
         };
 
-        vmm.add_serial_console()?;
         vmm.create_vcpus(&config.vcpu_config)?;
+        vmm.add_serial_console()?;
+
+        // We transiently define a mut `Cmdline` object here to pass for device creation
+        // (devices expect a `&mut Cmdline` to leverage the newly added virtio_mmio
+        // functionality), until we figure out how this fits with the rest of the vmm, which
+        // apparently does not explicitly use `Cmdline` structs. Will discuss and fix this
+        // somehow ASAP.
+        let mut cmdline = cmdline::Cmdline::new(4096);
+
+        if let Some(block_cfg) = config.block_config.as_ref() {
+            vmm.add_block_device(block_cfg, &mut cmdline)?;
+        }
+
+        vmm.kernel_cfg.cmdline.push_str(" ");
+        vmm.kernel_cfg.cmdline.push_str(cmdline.as_str());
```

Comment:
This is a bit ugly because if we have no additional devices there is no need to add the extra space to the cmdline. Can we for the time being move the cmdline changes in the previous if block?